Framework/API/inc/MantidAPI/ISpectrum.h

```diff
@@ -105,7 +105,7 @@ class DLLExport ISpectrum {
     HistogramData::Histogram histogram(std::forward<T>(data)...);
     // Check for the special case EventList, it only accepts histograms without
     // Y and E data.
-    checkHistogram(histogram);
+    checkAndSanitizeHistogram(histogram);
     mutableHistogramRef() = std::move(histogram);
```

Comment:
`checkAndSanitizeHistogram` takes care of that. For `EventList` it thows unless the `Y` and `E` in the `Histogram` are `nullptr`.